src/dht/dht_server.cc

```diff
@@ -597,7 +597,8 @@ void
 DhtServer::create_error(const DhtMessage& req, const rak::socket_address* sa, int num, const char* msg) {
   DhtMessage error;
 
-  if ((req[key_t].is_raw_bencode() || req[key_t].is_raw_string()) && req[key_t].as_raw_string().size() < 20)
+  if ((req[key_t].is_raw_bencode() && req[key_t].as_raw_bencode().size() < 20) ||
+		  (req[key_t].is_raw_string() && req[key_t].as_raw_string().size() < 20))
```

Comment:
The internets is somewhat ambiguous about what the key can be, so I guess we can make the assumption that it's always a raw_string since the rest of the code assumes it.